yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -454,10 +478,16 @@ private[spark] class Client(
      *   (3) whether to add these resources to the classpath
      */
     val cachedSecondaryJarLinks = ListBuffer.empty[String]
+    val files = sparkConf.get(FILES_TO_DISTRIBUTE).map(p => Utils.resolveURIs(p))
+      .orElse(sys.env.get("SPARK_YARN_DIST_FILES"))
```

Comment:
I think this could be done in SPARK-3374 as a separate patch to clean up all the internal env variables.